### PR TITLE
feat(deliveryconfig): initial scaffolding for deliveryconfig resource type

### PR DESCRIPTION
**Pull request by cfieber**

### CODE REVIEW

**Comment by cfieber** [comment]
just a first cut at setting up a plugin/handler for delivery-config definitions

File: keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt [hunk 1]
```diff
@@ -70,16 +65,3 @@ abstract class BaseModelParsingTest<out T> {
   }
 }
 
-fun Request.jsonResponse(
```
Comment (robfletcher) [inline keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt]: Where did this go?

File: keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt [hunk 2]
```diff
@@ -28,7 +23,7 @@ import strikt.assertions.isNotNull
 import java.net.URL
 import java.nio.charset.Charset
 
-abstract class BaseModelParsingTest<out T> {
+abstract class ModelParsingTestSupport<in S, out E>(serviceType: Class<S>) {
```
Comment (robfletcher) [inline keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt]: I’m surprised you even need the `in`

File: keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt (hunk 1 above)
Comment (cfieber) [reply keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt]: it wasn't used anywhere as far as I can see

File: keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt (hunk 2 above)
Comment (cfieber) [reply keel-retrofit-test-support/src/main/kotlin/com/netflix/spinnaker/keel/retrofit/model/ModelParsingTestSupport.kt]: the `in` was needed for the `call` signature of `S.() ->`

**Comment by robfletcher** [comment]
It looks fine, though. I don't see any issues with the Kotlin or testing.

**Pull request merged**